src/main/java/me/skulduggerry/gameoflife/converter/GenerationConverter.java

```diff
@@ -0,0 +1,40 @@
+package me.skulduggerry.gameoflife.converter;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.springframework.lang.NonNull;
+import org.springframework.stereotype.Component;
+
+import me.skulduggerry.gameoflife.model.Dimension;
+import me.skulduggerry.gameoflife.model.TransferGeneration;
+import me.skulduggerry.gameoflife.model.WorkGeneration;
+
+@Component
+public class GenerationConverter {
+
+    public WorkGeneration toWorkGeneration(@NonNull TransferGeneration transferGeneration) {
+        WorkGeneration.Cell[][] cells = transferGeneration.fieldData()
+                                                          .stream()
+                                                          .map(line -> Arrays.stream(line.split(""))
+                                                                             .map(cell -> cell.equals("x"))
+                                                                             .map(WorkGeneration.Cell::new)
+                                                                             .toArray(WorkGeneration.Cell[]::new))
+                                                          .toArray(WorkGeneration.Cell[][]::new);
+
+        WorkGeneration.getInstance().setNewData(transferGeneration.dimension(), cells);
```

Comment:
This method should only return the converted generation - and should not change the Singleton instance.

Reply:
Will be fixed in a coming version by using different data model